### PR TITLE
[ROCKETMQ-158]Remove logback dependency for rocketmq-tools

### PR DESCRIPTION
**Pull request by Jaskey**
jira: https://issues.apache.org/jira/browse/ROCKETMQ-158

### CODE REVIEW

**Comment by lizhanhui** [comment]
The logic here shares much similar logic with ClientLogger class, can we extract the common part into a reusable method?

**Comment by Jaskey** [comment]
@lizhanhui 

So do you agree this is indeed an issue that needed to be solved? If yes,  I will try doing that.

But actually it is a little different from the clientLogger since clientlogger will use the resources file in the classpath by default while tools admin does not have config file in the classpath.

**Comment by lizhanhui** [comment]
>  So do you agree this is indeed an issue that needed to be solved? If yes, I will try doing that.

Changing specific dependency to a neutral logging framework is good to have as application developer may integrate tool module to their operation management system.

 > But actually it is a little different from the clientLogger since clientlogger will use the resources file in the classpath by default while tools admin does not have config file in the classpath.

I know this, but it does not look like a blocking issue.

**Comment by Jaskey** [comment]
@lizhanhui 

please review the updated pr

**Comment by lizhanhui** [comment]
The conf folder has been moved to the distribution module, please rebase this PR on top of develop branch.  I'll +1 thereafter.

File: tools/src/main/java/org/apache/rocketmq/tools/command/MQAdminStartup.java [hunk 1]
```diff
@@ -196,22 +196,11 @@ private static void initLog() {
...
-                Class<?> domconfigurator;
-                Object domconfiguratorobj;
-                domconfigurator = Class.forName("org.apache.log4j.xml.DOMConfigurator");
-                domconfiguratorobj = domconfigurator.newInstance();
-                Method configure = domconfiguratorobj.getClass().getMethod("configure", String.class);
-                configure.invoke(domconfiguratorobj, rocketmqHome + "/conf/log4j_tools.xml");
+                final String logfjConfigPath = rocketmqHome + "/conf/log4j_tools.xml";
+                LogUtils.configLog4j(logfjConfigPath,null);
             } else if (classType.getName().equals("ch.qos.logback.classic.LoggerContext")) {
-                Class<?> joranConfigurator;
```
Comment (vongosling) [inline tools/src/main/java/org/apache/rocketmq/tools/command/MQAdminStartup.java]: Could we extract this log loader logic into one place

Comment (lizhanhui) [inline tools/src/main/java/org/apache/rocketmq/tools/command/MQAdminStartup.java]: Yes, we may wrap here  further @Jaskey

Comment (Jaskey) [inline tools/src/main/java/org/apache/rocketmq/tools/command/MQAdminStartup.java]: @vongosling the updated pr has been extract two logger config into util function. Do you mean the if else code snippet is need to wrap too?

**Comment by Jaskey** [comment]
@lizhanhui please review the updated pr, log4j_tools.xml has been move to distributions.

File: common/src/main/java/org/apache/rocketmq/common/utils/LogUtils.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import org.slf4j.ILoggerFactory;
+import org.slf4j.LoggerFactory;
+
+public class LogUtils {
+
+    /**
+     * config logback dymatically from provided file path, otherwise try to config from resource file
```
Comment (lizhanhui) [inline common/src/main/java/org/apache/rocketmq/common/utils/LogUtils.java]: Typo here: dymatically --> dynamically

File: common/src/main/java/org/apache/rocketmq/common/utils/LogUtils.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                doConfigure.invoke(joranConfiguratoroObj, url);
+            }
+        } else {
+            Method doConfigure = joranConfiguratoroObj.getClass().getMethod("doConfigure", String.class);
+            doConfigure.invoke(joranConfiguratoroObj, logConfigFilePath);
+        }
+    }
+
+    /**
+     * config log4j dymatically from provided file path, otherwise try to config from resource file. Notice that the log4j configuration file should be in xml format
```
Comment (lizhanhui) [inline common/src/main/java/org/apache/rocketmq/common/utils/LogUtils.java]: dymatically same as above.

**Comment by lizhanhui** [comment]
+1

File: common/src/main/java/org/apache/rocketmq/common/utils/LogUtils.java (hunk 2 above)
Comment (Jaskey) [inline common/src/main/java/org/apache/rocketmq/common/utils/LogUtils.java]: Thank you! I will update the commit soon.

File: common/src/main/java/org/apache/rocketmq/common/utils/LogUtils.java (hunk 3 above)
Comment (Jaskey) [inline common/src/main/java/org/apache/rocketmq/common/utils/LogUtils.java]: Fixed, thank you!

**Comment by Jaskey** [comment]
@zhouxinyu @shroman @vongosling 
what's your advice guys?

**Comment by vongosling** [comment]
+1

**Comment by zhouxinyu** [comment]
Hi @Jaskey 

Sorry to reply late, could you please resolve the conflicts, so we could merge this PR quickly.

**Comment by vongosling** [comment]
@Jaskey Could you pick up this pr

**Comment by Jaskey** [comment]
@zhouxinyu @vongosling , conflicts have been resolved, please review again. thank you

**Comment by vongosling** [comment]
@Jaskey I will close the pr, if you happened to the same question, please let me know.

**Pull request closed without merging**